### PR TITLE
schemas: add REST API endpoints for secret links

### LINKED ISSUE #436: versioning: implement /records/:id/access/links endpoint
**Goal**

Implement an initial access endpoint

**Tasks**

- Integrate into the existing resource
- Call ``service.update_permissions()``
- The ``:id`` should work with any id of the versions, so might need a special record, or ``update_permissions()`` should only touch ``record.parent``
- Direct index the record which id is being used, and reindex all previous versions in the background (send high priority task)

### PR DESCRIPTION
**Pull request by max-moser**
closes #436 
also introduces a new marshmallow schema, RDMParentSchema
requires inveniosoftware/invenio-drafts-resources/pull/116

### CODE REVIEW

File: invenio_rdm_records/services/services.py [hunk 1]
```diff
@@ -21,6 +29,283 @@ class RDMRecordService(RecordDraftService):
...
+
+    def link_result_list(self, *args, **kwargs):
+        """Create a new instance of the resource list."""
+        return self.config.link_result_list_cls(*args, **kwargs)
+
+    @property
+    def schema_parent(self):
+        """Schema for parent records."""
+        # TODO move to RecordDraftService?
+        return MarshmallowServiceSchema(self, schema=self.config.schema_parent)
```
Comment (max-moser) [inline invenio_rdm_records/services/services.py:44]: would it make sense to lift this up into RecordDraftService?

File: invenio_rdm_records/services/services.py [hunk 2]
```diff
@@ -21,6 +29,299 @@ class RDMRecordService(RecordDraftService):
...
+        )
+
+    def read_secret_link(
+        self,
+        id_,
+        identity,
+        link_id,
+        links_config=None,
+    ):
+        """Create a secret link for a record (resp. its parent)."""
```
Comment (lnielsen) [inline invenio_rdm_records/services/services.py:215]: Duplicated docstring a couple of times.

File: invenio_rdm_records/services/services.py [hunk 3]
```diff
@@ -21,6 +29,299 @@ class RDMRecordService(RecordDraftService):
...
+        return MarshmallowServiceSchema(self, schema=self.config.schema_parent)
+
+    @property
+    def schema_secret_link(self):
+        """Schema for secret links."""
+        return MarshmallowServiceSchema(
+            self, schema=self.config.schema_secret_link
+        )
+
+    def _get_record_and_parent_by_id(self, id_):
```
Comment (lnielsen) [inline invenio_rdm_records/services/services.py:53]: This one should probably also live in drafts-resources

File: invenio_rdm_records/services/services.py [hunk 4]
```diff
@@ -21,6 +29,299 @@ class RDMRecordService(RecordDraftService):
...
+
+            elif expires_at and expires_at < datetime.utcnow():
+                raise ValidationError(
+                    message=_("Expiration date must be set to the future"),
+                    field_name="expires_at",
+                )
+
+        return expires_at
+
+    def _index_related_records(self, record, parent):
```
Comment (lnielsen) [inline invenio_rdm_records/services/services.py:126]: Probably this should live in drafts-resources as all the logic related to parents is there.

**Review by lnielsen — COMMENTED** [review]
Looks good.....just some minor comments

File: invenio_rdm_records/services/result_items.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
```
[the remark this answers is not in the document]
Comment (max-moser) [reply invenio_rdm_records/services/result_items.py:1]: yes, i could've reused the `ResultItem` (but it's core property would've still been called `self._record`), but the `ResultList` was just way too different to really reuse

File: invenio_rdm_records/services/schemas/parent/__init__.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+from .access import ParentAccessSchema
+
+
+class RDMParentSchema(ParentSchema, FieldPermissionsMixin):
+    """Record schema."""
+
+    class Meta:
+        """Meta class."""
+
+        # TODO: RAISE instead!
```
[the remark this answers is not in the document]
Comment (max-moser) [reply invenio_rdm_records/services/schemas/parent/__init__.py:25]: nah, that just came for free with ^C ^V
and i figured that when the original TODO is done, this shouldn't be overlooked (thus, i kept the comment)

**Comment by max-moser** [comment]
if the tests are failing, that's likely due to the removal of the config in `resources.py`:
```python
config_name = "RDM_RECORDS_BIBLIOGRAPHIC_PARENT_RECORD_LINKS_CONFIG"
default_config = config.RDMParentRecordLinksResourceConfig
```

File: invenio_rdm_records/resources/resources.py [hunk 7]
```diff
@@ -106,3 +107,69 @@ class RDMDraftFilesActionResource(DraftFileActionResource):
...
     default_config = config.RDMDraftFilesActionResourceConfig
+
+
+#
+# Parent Record Links
+#
+class RDMParentRecordLinksResource(RecordResource):
+    """Secret links resource."""
+
+    config_name = "RDM_RECORDS_BIBLIOGRAPHIC_PARENT_RECORD_LINKS_CONFIG"
```
[the remark this answers is not in the document]
Comment (max-moser) [reply invenio_rdm_records/resources/resources.py:118]: re-added the config lines to make the tests pass for now, sorry! :disappointed:

**Pull request merged**